### ISSUE TITLE
latest webapp (main branch) only works in Chrome with an incognito window

### ISSUE THREAD

**Issue by davehorton** [jambonz/jambonz-webapp#170]
Note sure why @kitajchuk @EgleHelms  but when I run the latest webapp on a fully deployed server, when I go to the URL which is hosting jambonz I am immediately brought to a blank screen and the url is http:://<url>/internal/accounts instead of the login screen:

![image](https://user-images.githubusercontent.com/1884973/211177551-c8f8277b-2552-48c0-8e4b-1c195af9a43f.png)

On the other hand, if I am in a Chrome incognito window, it works
![image](https://user-images.githubusercontent.com/1884973/211177605-602dbe6d-4a36-46a0-9136-96b0a406afb7.png)

I can provide you access to the jambonz.one server to investigate if needed.

**Comment by kitajchuk** [comment]
> Note sure why @kitajchuk @EgleHelms but when I run the latest webapp on a fully deployed server, when I go to the URL which is hosting jambonz I am immediately brought to a blank screen and the url is http::///internal/accounts instead of the login screen:
> 
> ![image](https://user-images.githubusercontent.com/1884973/211177551-c8f8277b-2552-48c0-8e4b-1c195af9a43f.png)
> 
> On the other hand, if I am in a Chrome incognito window, it works ![image](https://user-images.githubusercontent.com/1884973/211177605-602dbe6d-4a36-46a0-9136-96b0a406afb7.png)
> 
> I can provide you access to the jambonz.one server to investigate if needed.

Did you try clearing your local and session storage?

**Comment by kitajchuk** [comment]
> > Note sure why @kitajchuk @EgleHelms but when I run the latest webapp on a fully deployed server, when I go to the URL which is hosting jambonz I am immediately brought to a blank screen and the url is http::///internal/accounts instead of the login screen:
> > ![image](https://user-images.githubusercontent.com/1884973/211177551-c8f8277b-2552-48c0-8e4b-1c195af9a43f.png)
> > On the other hand, if I am in a Chrome incognito window, it works ![image](https://user-images.githubusercontent.com/1884973/211177605-602dbe6d-4a36-46a0-9136-96b0a406afb7.png)
> > I can provide you access to the jambonz.one server to investigate if needed.
> 
> Did you try clearing your local and session storage?

That’s probably it because it worked for me and I’ve never been to that url before.

**Comment by davehorton** [comment]
I'm guessing you are right.  I will clear my local and session storage.....but should I have to in order to log in?  i.e. a "normal" user would not figure this out I dont think...

**Comment by kitajchuk** [comment]
> I'm guessing you are right. I will clear my local and session storage.....but should I have to in order to log in? i.e. a "normal" user would not figure this out I dont think...

No I’m not sure why this is happening. Maybe there is a bad redirect now with the user stuff? Did this happen before with the new app prior to the user changes?

**Comment by davehorton** [comment]
hm, cleared cache for the last 7 days and still had a problem.  I will clear entirely and try

**Comment by davehorton** [comment]
cleared all my browsing data for all of history and I still cant login unless in incognito.  Even opening a new normal Chrome window gives me the same failure

**Comment by davehorton** [comment]
I don't see any http redirect, I simply see a 200 OK coming back with the index.html, then downloading the fonts and assets.js and the URL updates to /internal/accounts and the screen is blank
![image](https://user-images.githubusercontent.com/1884973/211178215-f0c2da3f-c54b-4bfb-a0cc-bdfd8c76e34b.png)

with these errors in the console

![image](https://user-images.githubusercontent.com/1884973/211178231-b530870a-5784-4ea6-8897-797cbfb1e2bf.png)

**Comment by kitajchuk** [comment]
> I don't see any http redirect, I simply see a 200 OK coming back with the index.html, then downloading the fonts and assets.js and the URL updates to /internal/accounts and the screen is blank ![image](https://user-images.githubusercontent.com/1884973/211178215-f0c2da3f-c54b-4bfb-a0cc-bdfd8c76e34b.png)
> 
> with these errors in the console
> 
> ![image](https://user-images.githubusercontent.com/1884973/211178231-b530870a-5784-4ea6-8897-797cbfb1e2bf.png)

There you go. That would be the issue. Runtime error. What is causing it? Can you debug it further?

**Comment by kitajchuk** [comment]
> > I don't see any http redirect, I simply see a 200 OK coming back with the index.html, then downloading the fonts and assets.js and the URL updates to /internal/accounts and the screen is blank ![image](https://user-images.githubusercontent.com/1884973/211178215-f0c2da3f-c54b-4bfb-a0cc-bdfd8c76e34b.png)
> > with these errors in the console
> > ![image](https://user-images.githubusercontent.com/1884973/211178231-b530870a-5784-4ea6-8897-797cbfb1e2bf.png)
> 
> There you go. That would be the issue. Runtime error. What is causing it? Can you debug it further?

I’d bet it’s this line:

https://github.com/jambonz/jambonz-webapp/blob/93d313b21c9f488ce7f3b2074111504b398a0dfb/src/router/auth.tsx#L72

**Comment by davehorton** [comment]
Something to note is that I previously built the legacy server at this dns (jambonz.one) -- that was some time ago (months ago).  That server, when running, did not use jwt auth since that was only added to the main branch recently.  Could that somehow be related?

**Comment by davehorton** [comment]
In this case, I am never getting to the login screen.  Should we catch and fail on that error somehow so as to bring up the login screen so the person can re-login and get a proper jwt?

**Comment by kitajchuk** [comment]
> In this case, I am never getting to the login screen. Should we catch and fail on that error somehow so as to bring up the login screen so the person can re-login and get a proper jwt?

See my above comment. But ya, we need to examine our new code and properly catch this case and boot you out. I can take a look at that tomorrow if that works.

**Comment by kitajchuk** [comment]
> > In this case, I am never getting to the login screen. Should we catch and fail on that error somehow so as to bring up the login screen so the person can re-login and get a proper jwt?
> 
> See my above comment. But ya, we need to examine our new code and properly catch this case and boot you out. I can take a look at that tomorrow if that works.

Or even if you had one of the old, non jwt tokens, in storage and hit this route this error could happen too probably because the token syntax is different and it cannot split on a “.” that doesn’t exist.

**Comment by davehorton** [comment]
> ..I can take a look at that tomorrow if that works.

That would be awesome!

**Comment by kitajchuk** [comment]
@davehorton This is also a bad piece of code here. Our components should not be pulling the user like this. They need to get them from the store… I’m pretty confident I can reproduce this error locally so I can fix this and do another audit and make sure we’re always using safe syntax when working with the logged in user data.

https://github.com/jambonz/jambonz-webapp/blob/93d313b21c9f488ce7f3b2074111504b398a0dfb/src/containers/internal/views/accounts/index.tsx#L24

**Comment by kitajchuk** [comment]
@davehorton my last hunch was correct. fixed with this PR: https://github.com/jambonz/jambonz-webapp/pull/172

**Comment by davehorton** [comment]
Thanks so much!

**Comment by davehorton** [comment]
verified the fix resolved my issue.

**Comment by kitajchuk** [comment]
> verified the fix resolved my issue.

Excellent!

**Comment by EgleHelms** [comment]
@kitajchuk @davehorton 
this fix makes the bug of logging out as an account user and then logging in as admin throw this error since the state is not updated in time. So we did the same as for the create-password form read token directly instead of the state. I thought this was actually agreed on in the Scoping PR we merged.

![image](https://user-images.githubusercontent.com/63853378/211516919-83bd0f28-0c6e-4896-b428-582d50711b7c.png)
